hibernate-reactive-core/src/test/java/org/hibernate/reactive/ReactiveMultitenantNoResolverTest.java

```diff
@@ -266,74 +270,80 @@ public void testOpenStatelessSessionWithTenantWithMutiny(TestContext context) {
 
 	@Test
 	public void testOpenSessionThrowsExceptionWithoutTenant(TestContext context) {
-		thrown.expectCause( isA( HibernateException.class ) );
-		thrown.expectMessage( "no tenant identifier" );
-
-		test( context, openSession().thenCompose( this::selectCurrentDB ) );
+		test( context, ReactiveAssertions.assertThrown( HibernateException.class,
+								getSessionFactory().withSession( this::selectCurrentDB ) )
+						.thenAccept( exception -> Assertions.assertThat( exception.getMessage() )
+								.contains( "no tenant identifier" ) ) );
 	}
 
 	@Test
 	public void testWithSessionThrowsExceptionWithoutTenant(TestContext context) {
-		thrown.expectCause( isA( HibernateException.class ) );
-		thrown.expectMessage( "no tenant identifier" );
-
-		test( context, getSessionFactory().withSession( this::selectCurrentDB ) );
+		test( context, ReactiveAssertions.assertThrown( HibernateException.class, getSessionFactory().withSession( this::selectCurrentDB ) )
+				.thenAccept( exception -> Assertions.assertThat( exception.getMessage() ).contains( "no tenant identifier" ) ) );
 	}
 
 	@Test
 	public void testWithTransactionThrowsExceptionWithoutTenant(TestContext context) {
-		thrown.expectCause( isA( HibernateException.class ) );
-		thrown.expectMessage( "no tenant identifier" );
-
-		test( context, getSessionFactory().withTransaction( (s, t) -> selectCurrentDB( s ) ) );
+		test(context, ReactiveAssertions.assertThrown( HibernateException.class, getSessionFactory().withTransaction( this::selectCurrentDB ) )
+				.thenAccept( exception -> Assertions.assertThat( exception.getMessage() ).contains( "no tenant identifier" ) ) );
 	}
 
 	@Test
 	public void testWithStatelessTransactionThrowsExceptionWithoutTenant(TestContext context) {
-		thrown.expectCause( isA( HibernateException.class ) );
-		thrown.expectMessage( "no tenant identifier" );
-
-		test( context, getSessionFactory().withStatelessTransaction( (s, t) -> selectCurrentDB( s ) ) );
+		test( context, ReactiveAssertions.assertThrown( HibernateException.class, getSessionFactory().withStatelessTransaction( this::selectCurrentDB ) )
+				.thenAccept( exception -> Assertions.assertThat( exception.getMessage() ).contains( "no tenant identifier" ) ) );
 	}
 
 	@Test
 	public void testOpenSessionThrowsExceptionWithoutTenantWithMutiny(TestContext context) {
-		thrown.expect( HibernateException.class );
-		thrown.expectMessage( "no tenant identifier" );
-
-		test( context, openMutinySession().call( this::selectCurrentDB ) );
+		test( context, ReactiveAssertions.assertThrown(
+				HibernateException.class, openMutinySession().invoke( this::selectCurrentDB ) )
+				.onFailure().recoverWithUni( exception -> {
```

Comment:
This code will never get code if there is an exception (like we expect)

Reply:
refactored and fixed